### ISSUE TITLE
Change CI systems

### ISSUE THREAD

**Issue by dbt** [factoriotools/factorio-docker#363]
travis-ci.org is shutting down soon and it looks like it's already falling behind on running new builds. Free alternatives include https://circleci.com/open-source/ and Github Actions (I'm sure there are others).

**Comment by ProfBorkensnoot** [comment]
Thanks for getting ahead of me on this. I was a contributor on another factorio docker and that didn't require a slow CI pipeline to update on DockerHub.
I'm happy to contribute server capacity if it means speeding things up :)

**Comment by SuperSandro2000** [comment]
I am going to replace travis with github actions. I hoped the releases would slow down a bit so that it wouldn't be a problem.

**Comment by jswny** [comment]
I think a good improvement to keep in mind while migrating is to only build/push images for versions that actually got bumped. For example, when bumping the experimental version, it looks like every version still gets built even though only `1.1` actually needs to be built. This should be fairly easy to do, one way would just be to grab the list of files changed in the last commit (seems to me like your bot bumps the Dockerfile in the correct version directory), and then grep it to figure out which ones you need to build. Something like this for example:

```
git diff-tree --no-commit-id --name-only -r HEAD | grep -q 1.1/
```

**Comment by SuperSandro2000** [comment]
I won't do that because then all the other images are using old versions of the packages included.

**Comment by jswny** [comment]
Not really sure what you mean by that. The packages inside the Dockerfile?

**Comment by SuperSandro2000** [comment]
> The packages inside the Dockerfile?

Yes

**Comment by lucasreiners** [comment]
Thankyou for your work! If you need any help let me know, although I see you are already busy working on the migration :-)

**Comment by tflinz** [comment]
Still seeing the 1.19 building (stalled) on travis, fyi: https://travis-ci.org/github/factoriotools/factorio-docker/builds/756262796

(not sure if it should be cancelled and then the github-route triggered now that it is merged in)

**Comment by coreysabia** [comment]
@tflinz, @SuperSandro2000 went the GitHub Actions route, per recent commits. However after checking the docker hub, it looks like the description is getting updated (per one of the two CI jobs) but 1.1.19 is not getting pushed from the other CI job yet.

@SuperSandro2000 you might want to double check your CI pipeline.

**Comment by tflinz** [comment]
Heh, yes, that's why I commented "the github-route triggered now that it is merged in" ;)

**Comment by SuperSandro2000** [comment]
> it looks like the description is getting updated (per one of the two CI jobs) but 1.1.19 is not getting pushed from the other CI job yet.

The description is pushed with every commit and the 1.1.19 tag is only released when a tag 1.1.19 or 1.1.19-[1-9*] is pushed.
